### PR TITLE
fix #129296 keep authority, query, fragment when navigating simple file dialog

### LINKED ISSUE #129296: When a local workspace root uses a FileSystemProvider where `authority` element of uri is significant, the SimpleFileDialog presented by `File\Open File` cannot navigate into subfolders
Issue Type: <b>Bug</b>

The problem seems to lie in this code:
https://github.com/microsoft/vscode/blob/79e91942bc029ea93426a840e83495f9a5161d12/src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts#L213-L224

When the workspace is a local one, `this.remoteAuthority` is undefined, so the returned uri has `authority: ''`

I intend to prepare a variant of the MemFS sample to demonstrate the problem.

VS Code version: Code - Insiders 1.59.0-insider (04485223ebc6a075f39966bf9b3e261359ac7818, 2021-07-23T19:21:12.120Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

### PR DESCRIPTION
**Pull request by gjsjohnmurray**
This PR fixes #129296

Test by using the fsprovider-sample branch referenced in the original issue.

### CODE REVIEW

File: src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts [hunk 1]
```diff
@@ -210,17 +210,17 @@ export class SimpleFileDialog {
...
+	private remoteUriFrom(path: string, hintUri?: URI): URI {
 		if (!path.startsWith('\\\\')) {
 			path = path.replace(/\\/g, '/');
 		}
-		const uri: URI = this.scheme === Schemas.file ? URI.file(path) : URI.from({ scheme: this.scheme, path });
-		// If the default scheme is file, then we don't care about the remote authority
-		const authority = uri.scheme === Schemas.file ? undefined : this.remoteAuthority;
+		const uri: URI = this.scheme === Schemas.file ? URI.file(path) : URI.from({ scheme: this.scheme, path, query: hintUri?.query, fragment: hintUri?.fragment });
+		// If the default scheme is file, then we don't care about the remote authority or the hint authority
+		const authority = uri.scheme === Schemas.file ? undefined : this.remoteAuthority ?? hintUri?.authority;
```
Comment (alexr00) [inline src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts:219]: Can you add parentheses here to indicate the intended evaluation order?

**Review by alexr00 — APPROVED** [review]
Thanks @gjsjohnmurray, this looks good. I will test and merge for the August milestone once we're wrapping up the July release.

File: src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts (hunk 1 above)
Comment (gjsjohnmurray) [reply src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts]: @alexr00 I pushed a change. Does it do what you want?

Comment (alexr00) [reply src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts]: Yes, that's exactly it.

**Pull request merged**